pkg/scheduler/framework/plugins/gangscheduling/gangscheduling.go

```diff
@@ -0,0 +1,282 @@
+	originalWorkload, modifiedWorkload, err := util.As[*schedulingapi.Workload](oldObj, newObj)
+	if err != nil {
+		return fwk.Queue, err
+	}
+
+	if pod.Spec.Workload == nil || pod.Namespace != modifiedWorkload.Namespace || pod.Spec.Workload.Name != modifiedWorkload.Name {
+		logger.V(5).Info("workload was added or updated but it doesn't match the target pod's workload",
+			"pod", klog.KObj(pod), "modifiedWorkload", klog.KObj(modifiedWorkload))
+		return fwk.QueueSkip, nil
+	}
+
+	if originalWorkload == nil {
+		logger.V(5).Info("workload was added and it matches the target pod's workload, which may make the pod schedulable",
+			"pod", klog.KObj(pod), "modifiedWorkload", klog.KObj(modifiedWorkload))
+		return fwk.Queue, nil
+	}
+
+	originalPolicy, originalOk := podGroupPolicy(originalWorkload, pod.Spec.Workload.PodGroup)
+	modifiedPolicy, modifiedOk := podGroupPolicy(modifiedWorkload, pod.Spec.Workload.PodGroup)
+	if originalOk && !modifiedOk || !originalOk && modifiedOk {
+		return fwk.Queue, fmt.Errorf("unexpected modification of workload pod groups")
+	}
+	if modifiedOk {
+		switch modifiedPolicy.Kind {
+		case schedulingapi.PodGroupPolicyKindDefault:
+			logger.V(5).Info("workload was updated and matching pod group has a Default policy, which may make the pod schedulable",
+				"pod", klog.KObj(pod), "originalWorkload", klog.KObj(originalWorkload), "modifiedWorkload", klog.KObj(modifiedWorkload))
+			return fwk.Queue, nil
+		case schedulingapi.PodGroupPolicyKindGang:
+			if originalPolicy.Gang.MinCount > modifiedPolicy.Gang.MinCount {
+				logger.V(5).Info("workload was updated and matching pod group's minCount is lower, which may make the pod schedulable",
+					"pod", klog.KObj(pod), "originalWorkload", klog.KObj(originalWorkload), "modifiedWorkload", klog.KObj(modifiedWorkload))
+				return fwk.Queue, nil
+			}
+		}
+	}
+
+	logger.V(5).Info("workload was updated, but its new policy is not making the pod schedulable",
+		"pod", klog.KObj(pod), "originalWorkload", klog.KObj(originalWorkload), "modifiedWorkload", klog.KObj(modifiedWorkload))
+	return fwk.QueueSkip, nil
+}
+
+// PreEnqueue checks if the pod belongs to a gang and, if so, whether the gang has met its MinCount of available pods.
+// If not, the pod is rejected until more pods arrive.
+func (pl *GangScheduling) PreEnqueue(ctx context.Context, pod *v1.Pod) *fwk.Status {
+	if !pl.enabled || pod.Spec.Workload == nil {
+		return nil
+	}
+
+	namespace := pod.Namespace
+	workloadRef := pod.Spec.Workload
+
+	workload, err := pl.workloadLister.Workloads(namespace).Get(workloadRef.Name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// The pod is unschedulable until its Workload object is created.
+			return fwk.NewStatus(fwk.UnschedulableAndUnresolvable, "waiting for pods's workload to appear in scheduling queue")
+		}
+		return fwk.AsStatus(fmt.Errorf("failed to get workload %s/%s: %w", namespace, workloadRef.Name, err))
```

Comment:
Can we log `err`, and return `"failed to get workload %s/%s" only? (to try making the error message as consistent as possible, as we discussed in the PreEnqueue error visibility KEP)